Sources/SwiftParser/Specifiers.swift

```diff
@@ -797,20 +798,53 @@ extension Parser {
         unexpected.append(misspelledThrows)
         if effectSpecifiers?.throwsClause == nil {
           if matchedSubset == S.CorrectThrowsTokenKinds.self {
-            synthesizedThrows = missingToken(spec)
+            let synthesizedThrows = missingToken(spec)
+            var lookahead = self.lookahead()
+            guard lookahead.consume(if: .leftParen) != nil,
+              lookahead.canParseType()
+            else {
+              synthesizedThrowsClause = RawThrowsClauseSyntax(
+                throwsSpecifier: synthesizedThrows,
+                leftParen: nil,
+                type: nil,
+                rightParen: nil,
+                arena: self.arena
+              )
+              break
+            }
+
+            let leftParen = self.consume(if: .leftParen)!
```

Comment:
Let’s make this `self.expect`. That way we avoid the force unwrap.

```suggestion
            let leftParen = self.expect(.leftParen)
```

And for symmetry, I would also use `expect` to get the right parenthesis.

Reply:
In my current design, a thrown type missing the right parenthesis is allowed to recover, do you think such recovery should be disallowed?

Reply:
It should be able to recover with `self.expect` as well. The only difference is whether `rightParen` is set to `nil` or a missing right parenthesis.